modular_skyrat/modules/reagent_forging/code/crafting_bench.dm

```diff
@@ -0,0 +1,242 @@
+	)
+
+/obj/structure/reagent_crafting_bench/examine(mob/user)
+	. = ..()
+	if(goal_name)
+		. += span_notice("Goal Item: [goal_name]")
+		. += span_notice("When you have the necessary materials, begin hammering!<br>")
+		if(required_chain)
+			. += span_warning("[chain_amount]/[required_chain] chains stored.")
+		if(required_plate)
+			. += span_warning("[plate_amount]/[required_plate] plates stored.")
+		if(required_coil)
+			. += span_warning("[coil_amount]/[required_coil] coils stored.")
+	if(length(contents))
+		. += span_notice("Held Item: [contents[1]]")
+
+/obj/structure/reagent_crafting_bench/update_appearance(updates)
+	. = ..()
+	icon_state = "crafting_bench_[length(contents) ? "full" : "empty"]"
+
+//when picking a design or clearing a design
+/obj/structure/reagent_crafting_bench/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(length(contents))
+		var/obj/item/moving_item = contents[1]
+		user.put_in_hands(moving_item)
+		balloon_alert(user, "item retrieved!")
+		update_appearance()
+		return
+	if(goal_item_path)
+		clear_required()
+		balloon_alert_to_viewers("table cleared!")
+		update_appearance()
+		return
+	var/target_choice = tgui_input_list(user, "Which item would you like to craft?", "Crafting Choice", allowed_choices)
+	if(!target_choice)
+		balloon_alert(user, "no choice made!")
+		return
+	goal_name = target_choice
+	goal_item_path = allowed_choices[target_choice]
+	switch(target_choice)
+		if("Chain Helmet")
+			required_chain = 5
+		if("Chain Armor")
+			required_chain = 6
+		if("Chain Gloves")
+			required_chain = 4
+		if("Chain Boots")
+			required_chain = 4
+		if("Plated Boots")
+			required_plate = 4
+		if("Horseshoes")
+			required_chain = 4
+		if("Ring")
+			required_chain = 2
+		if("Collar")
+			required_chain = 3
+		if("Handcuffs")
+			required_chain = 10
+		if("Borer Cage")
+			required_plate = 6
+		if("Pavise Shield")
+			required_plate = 8
+		if("Buckler Shield")
+			required_plate = 5
+		if("Coil")
+			required_chain = 2
+	if(!required_hits)
+		required_hits = (required_chain * 2) + (required_plate * 2) + (required_coil * 2)
+	balloon_alert(user, "choice made!")
+	update_appearance()
+
+/obj/structure/reagent_crafting_bench/proc/clear_required()
+	required_hits = 0
+	current_hits = 0
+	goal_item_path = null
+	goal_name = null
+	required_chain = 0
+	required_plate = 0
+	required_coil = 0
+
+/obj/structure/reagent_crafting_bench/proc/check_required_materials()
+	if(chain_amount < required_chain)
+		return FALSE
+	if(plate_amount < required_plate)
+		return FALSE
+	if(coil_amount < required_coil)
+		return FALSE
+	return TRUE
+
+//when inserting the materials
+/obj/structure/reagent_crafting_bench/attackby(obj/item/attacking_item, mob/user, params)
+	update_appearance()
+	//if we are attacking with a hammer and we have a goal in mind!
+	if(istype(attacking_item, /obj/item/forging/hammer))
+		playsound(src, 'modular_skyrat/modules/reagent_forging/sound/forge.ogg', 50, TRUE)
+		if(length(contents))
```

Comment:
early return here? "nothing on the table!"?

Reply:
cant have an early return because of later code requiring an empty table